### PR TITLE
ClassifyElement: more tests

### LINKED ISSUE #25: ClassifyElement: Test more edge cases
We should add more tests, specifically we need tests that contain more output branches, have longer to compute split functions, have multiple layers of channels. etc etc.

### PR DESCRIPTION
**Pull request by ddowl**
I noticed that certain router configs would either panic or not run at
all when not referencing the last `provider` in a `ClassifyElementLink`
via `pop`. That might require more investigation as we write our POC
integration tests.

resolves #25

### CODE REVIEW

File: src/api/classify_element.rs [hunk 1]
```diff
@@ -318,46 +337,149 @@ mod tests {
...
+    }
+
+    #[test]
+    fn fizz_buzz_to_even_odd() {
+        let default_channel_size = 10;
+        let packet_generator = immediate_stream(0..=30);
+
+        let fizz_buzz_elem = ClassifyFizzBuzzElement { id: 0 };
+
+        let mut fizz_buzz_elem_link: ClassifyElementLink<ClassifyFizzBuzzElement> =
```
Comment (eeverett6) [inline src/api/classify_element.rs:437]: Is the fully specified type needed here because we have two different ClassifyElementLinks?

**Review by bweis — APPROVED** [review]
LGTM
